### DISCUSSION TITLE
How to update UserControl property

### DISCUSSION THREAD

**Discussion by Umenokin** [AvaloniaUI/Avalonia#9094] [Q&A]
I have a simple component `PanelHeader.axaml`:
```
<UserControl xmlns="https://github.com/avaloniaui"
             xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
             xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
             xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
             mc:Ignorable="d" d:DesignWidth="800" d:DesignHeight="450"
             x:Class="Flexidy.Studio.Views.Common.Panel.PanelHeader"
             DataContext="{Binding RelativeSource={RelativeSource Self}}">
    
    <StackPanel Classes="panelHeader">
        <Grid ColumnDefinitions="*, Auto" RowDefinitions="*, 1">
            <TextBlock Grid.Row="0" Grid.Column="0" Text="{Binding Title}" />
            <StackPanel Grid.Row="0" Grid.Column="1" Orientation="Horizontal">
                <TransitioningContentControl Content="{Binding Menu}"></TransitioningContentControl>
            </StackPanel>
        </Grid>
        <Rectangle Classes="divider" />
    </StackPanel>
</UserControl>
```

Where `Title` and `Menu` are properties of `PanelHeader.axaml.cs`. I implemented them from official documentation:
```
public static readonly DirectProperty<PanelHeader, string> TitleProperty =
    AvaloniaProperty.RegisterDirect<PanelHeader, string>(
      nameof(Title),
      o => o.Title,
      (o, v) => o.Title = v);
  
  public string Title
  {
    get { return _title; }
    set { SetAndRaise(TitleProperty, ref _title, value); }
  }
  
  private string _title = "";
```

Then, in another component I would use it like so:
```
<panels:PanelHeader DockPanel.Dock="Top" Title="Properties" />
```

At this point works fine because `Title` would never change, but if I make the title dynamic like in the example below it will never change:
```
<panels:PanelHeader DockPanel.Dock="Top" Title="{Binding EntityName}" />
<TextBlock Text="{Binding EntityName}"></TextBlock>
```

I am sure that `EntityName` is not an issue because TextBlock in the example above will be updated once EntityName has changed while my PanelHeader does not react to `Title` changes.

#### UPDATE: If I add the log to Title setter it does not look like that it will be called once `EntityName` has changed:
```
public string Title
  {
    get { return _title; }
    set
    {
      Console.WriteLine($"PanelHeader set title: {value}");
      SetAndRaise(TitleProperty, ref _title, value);
    }
  }
```
And code below will not call setter:
```
<panels:PanelHeader DockPanel.Dock="Top" Title="{Binding EntityName}" />
```

Thanks!

**Comment by Umenokin** [comment]
Found [Issue](https://github.com/AvaloniaUI/Avalonia/issues/4680) that might be what I am looking for:

> @derekantrican you should override OnPropertyChanged method to handle it, or subscribe from Rx observable.
Ad @ahopper said, WPF and UWP should have same behavior.

**Comment by rabbitism** [comment]
I usually don't set DataContext to control itself. instead I will use  `Text="{Binding $parent[UserControl].Title}` for such a purpose.

**Reply by Umenokin** [reply]
Thanks, @rabbitism! It does not solve my problem but I like your approach because it is much more explicit.
